tests/model/selection.js

```diff
@@ -248,6 +248,24 @@ describe( 'Selection', () => {
 		} );
 	} );
 
+	describe( 'focus', () => {
+		it( 'should return first end start', () => {
+			selection.addRange( range );
+
+			expect( selection.focus.isEqual( range.end ) ).to.be.true;
+		} );
+
+		it( 'should return first range start when backward', () => {
```

Comment:
`focus` does not return positions from first range. `focus` is connected with last added range (not "first" or "last" range in position relation meaning). No matter if selection is backward or not, `focus` and `anchor` are connected with last range (so is `isBackward` property).

Those tests should have three ranges added, which are "mixed" in position relation meaning (i.e. range 2-4, range 4-6 and range 1-2). In such test `focus` should be `1` for normal direction and `2` for backward direction.